tests/test_parser.py

```diff
@@ -2085,3 +2085,76 @@ def test_string_attribute_reference_5():
     assert instance1.name == "x"
     assert str(instance1.locatable_name) == "x"
     assert instance1.locatable_name.location == Range("test", 5, 1, 5, 2)
+
+
+@pytest.mark.parametrize_any(
+    "snippet",
+    [
+        "mymod.MyEntity()",
+        "mymod.submod.MyEntity()",
+        "mymod.submod.MyEntity(x=1)",
+        "mymod.submod.MyEntity(**dct)",
+        "entity Child extends mymod.MyEntity: end",
+        "SomeEntity.my [1] -- mymod.MyEntity",
+    ],
+)
+def test_entity_ref_err_dot(snippet: str) -> None:
+    """
+    Verify that an attempt to access an entity in a qualified manner with '.' instead of '::' results in an appropriate
+    exception.
+
+    :param snippet: Snippet that is expected to produce this error for an entity named `MyEntity`.
+    """
+    with pytest.raises(ParserException, match="`MyEntity` looks like an entity but was accessed with '.'"):
+        parse_code(snippet)
+
+
+def test_entity_ref_err_dot_full_msg() -> None:
+    with pytest.raises(
+        ParserException,
+        match=re.escape(
+            "Syntax error: `MyEntity` looks like an entity but was accessed with '.' (`mymod.submod.MyEntity`)."
+            " To access an entity in a namespace, use '::' instead: `mymod::submod::MyEntity` (test:2:5)"
+        ),
+    ):
+        parse_code(
+            """
+x = mymod.submod.MyEntity()
+            """
+        )
+
+
+@pytest.mark.parametrize_any(
+    "snippet",
+    [
+        "mymod.my_plugin()",
+        "mymod.submod.my_plugin()",
+        "mymod.submod.my_plugin(1)",
+        "mymod.submod.my_plugin(x=1)",
+        "mymod.submod.my_plugin(**dct)",
+    ],
+)
+def test_plugin_call_err_dot(snippet: str) -> None:
+    """
+    Verify that an attempt to access a plugin in a qualified manner with '.' instead of '::' results in an appropriate
+    exception.
+
+    :param snippet: Snippet that is expected to produce this error for a plugin named `my_plugin`.
+    """
+    with pytest.raises(ParserException, match=r"can only call plugins but [a-z\.]*my_plugin looks like an attribute."):
+        parse_code(snippet)
+
+
+def test_plugin_call_err_dot_full_msg() -> None:
+    with pytest.raises(
+        ParserException,
+        match=re.escape(
+            "Syntax error: can only call plugins but mymod.submod.my_plugin looks like an attribute."
```

Comment:
1. quoting of the `mymod.submod.my_plugin` is not consistent with the previous 
2. why not make this error message more like the other?

Reply:
2. Because they appear in a different context: it's a plugin call vs any reference to an entity (constructor, index lookup, relation declaration, ...). That said, I'm not great at error messages. Perhaps I should include @bartv and @arnaudsjs for a review of the messages (the rest of the PR can be considered out of scope for the review).

Reply:
I do not understand the error message actually. This is the message you get when you use attribute access but with a plugin call "argument"?

Reply:
Yes. This one is for `std.print(1)` instead of `std::print(1)`. The other message is for `std.Entity()` instead of `std::Entity()`.

Reply:
Then I would say that:

`Plugins should be called with :: instead of .`

or even beter when the actual call was std.print:

`Plugins should be called with std::print instead of std.print`

You could add a second sentence that says something like:

`The . operator should only be used for attribute and relation access`

Reply:
I like Bart his suggestion

Reply:
I took Bart's suggestion and made it a bit more generic so I could go with Wouter's suggestion (after all) of making both messages more uniform. Let me know what you think.